componentLibraries/defaultLibrary/Connectivity/FMIWrapper.hpp

```diff
@@ -0,0 +1,1324 @@
+            
+            fmi2Status status;
+            
+           //Read inputs
+
+            std::map<fmi2ValueReference,double*>::iterator it;
+            for(it = mRealInputs.begin(); it != mRealInputs.end(); it++) {
+                status = fmi2SetReal(fmu, &it->first, 1, it->second);
+            }
+            for(it = mIntInputs.begin(); it != mIntInputs.end(); it++) {
+                int value = int(*it->second);
+                status = fmi2SetInteger(fmu, &it->first, 1, &value);
+            }
+            for(it = mBoolInputs.begin(); it != mBoolInputs.end(); it++) {
+                int value = int(*it->second);
+                status = fmi2SetBoolean(fmu, &it->first, 1, &value);
+            }
+     
+             //Take step
+             bool eventEncountered, terminateSimulation, earlyReturn;
+             double lastT;
+             status = fmi2DoStep(fmu, mTime-mTimestep, mTimestep, fmi3True);
+             if (status != fmi2OK) {
+                 stopSimulation("fmi2DoStep() failed, status = "+to_hstring(status));
+                 return;
+             }
+     
+             //Write outputs
+             for(it = mRealOutputs.begin(); it != mRealOutputs.end(); it++) {
+                 status = fmi2GetReal(fmu, &it->first, 1, it->second);
+             }
+             for(it = mIntOutputs.begin(); it != mIntOutputs.end(); it++) {
+                 int temp;
+                 status = fmi2GetInteger(fmu, &it->first, 1, &temp);
+                 (*it->second) = temp;
+             }
+             for(it = mBoolOutputs.begin(); it != mBoolOutputs.end(); it++) {
+                 int temp;
+                 status = fmi2GetBoolean(fmu, &it->first, 1, &temp);
+                 (*it->second) = temp;
+             }
+        }
+        else { //FMI 3
+            if(NULL == fmu) {
+                return;
+            }
+            fmi3Status status;
+            
+            //Read inputs
+            std::map<fmi3ValueReference,double*>::iterator itr;
+            for(itr = mFloat64Inputs.begin(); itr != mFloat64Inputs.end(); itr++) {
+                fmi3Float64 value = (fmi3Float64)(*itr->second);
+                status = fmi3SetFloat64(fmu, &itr->first, 1, &value, 1);
+            }
+            for(itr = mFloat32Inputs.begin(); itr != mFloat32Inputs.end(); itr++) {
+                fmi3Float32 value = (fmi3Float32)(*itr->second);
+                status = fmi3SetFloat32(fmu, &itr->first, 1, &value, 1);
+            }
+            std::map<fmi3ValueReference,double*>::iterator itb;
+            for(itb = mBoolInputs.begin(); itb != mBoolInputs.end(); ++itb) {
+                bool value = ((*itb->second) > 0.5);
+                status = fmi3SetBoolean(fmu, &itb->first, 1, &value, 1);
+            }
+            std::map<fmi3ValueReference,double*>::iterator iti;
+            for(iti = mInt64Inputs.begin(); iti != mInt64Inputs.end(); ++iti) {
+                fmi3Int64 value = (fmi3Int64)(*iti->second);
```

Comment:
for all the ints, do we need to take care of rounding issues ?
Like (intType)(realValue+0.5), to avoid converting 2.999999999999 to 2 instead of 3
Not sure if this is an issue

Reply:
Of course, you are right. I have not tried any FMUs with integer inputs yet. I will add `lround` for all integer inputs.